pkg/desktop/desktop.go

```diff
@@ -0,0 +1,156 @@
+package desktop
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/browser"
+	"github.com/stashapp/stash/pkg/database"
+	"github.com/stashapp/stash/pkg/logger"
+	"github.com/stashapp/stash/pkg/manager/config"
+	"github.com/stashapp/stash/pkg/utils"
+	"golang.org/x/term"
+)
+
+func Start() {
+	if IsDesktop() {
+		OpenURLInBrowser(false, "")
+		writeStashIcon()
+		startSystray()
+	}
+}
+
+// OpenURLInBrowser opens a browser to the Stash UI. Path can be an empty string for main page.
+func OpenURLInBrowser(force bool, path string) {
+	// This can be done before actually starting the server, as modern browsers will
+	// automatically reload the page if a local port is closed at page load and then opened.
+	c := config.GetInstance()
+	if force || (!c.GetNoBrowser() && IsDesktop()) {
+		serverAddress := getServerURL(path)
+
+		err := browser.OpenURL(serverAddress)
+		if err != nil {
+			logger.Error("Could not open browser: " + err.Error())
+		}
+	}
+}
+
+func SendNotification(title string, text string) {
+	if IsDesktop() {
+		c := config.GetInstance()
+		if c.GetNotificationsEnabled() {
+			sendNotification(title, text)
+		}
+	}
+}
+
+func IsDesktop() bool {
+	// Check if running under root
+	if os.Getuid() == 0 {
+		return false
+	}
+	// Check if stdin is a terminal
+	if term.IsTerminal(int(os.Stdin.Fd())) {
```

Comment:
Just double checking for this one and linux systems
In the current dev this wasn't checked so stash could open a browser window when run from a terminal
With this applied only if you double click and execute stash from an explorer the browser window and the notifications are working

Reply:
That's intended. I got annoyed with Stash taking over my terminal with a `links` window when I was testing a build over ssh